File: Server/route/userCTFContainer.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetUserCTFContainerRoute(r *gin.Engine) {
	userContainer := r.Group("/userCTFContainer")
	userContainer.Use(middleware.JWTAuthMiddleware())
	{
		userContainer.POST("/GetCTFContainerListByUser", api.GetCTFContainerListByUser)
		userContainer.POST("/CreateCTFContainerByUser", api.CreateCTFContainerByUser)
		userContainer.POST("/DestroyCTFContainerByUser", api.DestroyCTFContainerByUser)
		userContainer.POST("/DelayCTFContainerByUser", api.DelayCTFContainerByUser)
	}
}

File: Web/src/api/userCTFContainer.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetCTFContainerListByUser() {
  return apiClient.post("/userCTFContainer/GetCTFContainerListByUser").then(handleResponse).catch(handleError)
}

function CreateCTFContainerByUser(data) {
  return apiClient.post("/userCTFContainer/CreateCTFContainerByUser", data).then(handleResponse).catch(handleError)
}

function DestroyCTFContainerByUser(data) {
  return apiClient.post("/userCTFContainer/DestroyCTFContainerByUser", data).then(handleResponse).catch(handleError)
}

function DelayCTFContainerByUser(data) {
  return apiClient.post("/userCTFContainer/DelayCTFContainerByUser", data).then(handleResponse).catch(handleError)
}

const userCTFContainerApi = {
  GetCTFContainerListByUser,
  CreateCTFContainerByUser,
  DestroyCTFContainerByUser,
  DelayCTFContainerByUser
}

export default userCTFContainerApi
